src/OpenSearch/Namespaces/AbstractNamespace.php

```diff
@@ -31,15 +33,27 @@ abstract class AbstractNamespace
      */
     protected $transport;
 
+    protected EndpointFactoryInterface $endpointFactory;
+
     /**
      * @var callable
+     *
+     * @deprecated in 2.3.2 and will be removed in 3.0.0. Use $endpointFactory property instead.
      */
     protected $endpoints;
 
-    public function __construct(Transport $transport, callable $endpoints)
+    public function __construct(Transport $transport, callable|EndpointFactoryInterface $endpointFactory)
     {
         $this->transport = $transport;
+        if (is_callable($endpointFactory)) {
+            @trigger_error('Passing a callable as $endpointFactory param to ' . __METHOD__ . '() is deprecated in 2.3.2 and will be removed in 3.0.0. Pass an instance of \OpenSearch\EndpointFactoryInterface instead.', E_USER_DEPRECATED);
+            $endpoints = $endpointFactory;
+            $endpointFactory = new LegacyEndpointFactory($endpointFactory);
+        } else {
+            $endpoints = fn ($c) => $endpointFactory->getEndpoint('OpenSearch\\Endpoints\\' . $c);
```

Comment:
this should have the same deprecation trigger than in the client when building such a BC callable.

Reply:
Fixed :heavy_check_mark: